### PR TITLE
V7 Add the predefined interface type Throwable

### PR DESCRIPTION
**Pull request by RexJaeschke**

### CODE REVIEW

File: spec/15-interfaces.md [hunk 1]
```diff
@@ -220,6 +220,40 @@ Name  |   Purpose
 ----    |   -------
 `getIterator` | This instance method retrieves an iterator, which implements `Iterator` or `Traversable`. It throws an `Exception` on failure.
 
+###Interface `Throwable`
+
+This type is the base interface for the type of any object that can be thrown via a
+[throw statement](11-statements.md#the-throw-statement). A user-written class cannot
+implement `Throwable` directly. Instead, it must extend `Exception`.
```
Comment (nikic) [inline spec/15-interfaces.md]: `Exception` or `Error`. Both are user extensible :)

**Comment by RexJaeschke** [comment]
Added `Error` as suggested. Squash of 2 commits soon to some.

File: spec/15-interfaces.md [hunk 2]
```diff
@@ -220,6 +220,40 @@ Name  |   Purpose
...
+
+
+Name  |   Purpose
+----    |   -------
+`__toString`	| `string`; retrieves a string representation of the exception in some unspecified format
+`getCode`	| `int`; retrieves the exception code (as set by the constructor).
+`getFile`	| `string`; retrieves the name of the script where the exception was generated
+`getLine`	| `int`; retrieves the source line number in the script where the exception was generated
+`getMessage`	| `string`; retrieves the exception message
+`getPrevious`	| `Throwable`; retrieves the previous exception (as set by the constructor), if one exists; otherwise, `NULL`
```
Comment (nikic) [inline spec/15-interfaces.md]: This reads a bit weird moved to Throwable, as Throwable itself doesn't have a constructor.

Comment (RexJaeschke) [inline spec/15-interfaces.md]: Agreed. If I remove the two parenthetic comments re the constructor, will that suffice?

**Comment by nikic** [comment]
Merged via https://github.com/php/php-langspec/commit/aeda4495736b75d0005c87a9336a83132863afab with the parenthetical removed per your suggestion.

**Pull request closed without merging**